### PR TITLE
Alternative implementation using existing `bits` and `bytes` attributes

### PR DESCRIPTION
**Pull request by sharksforarms**
Hey @samuelsleight I thought I'd propose some changes to your PR based on my comment in https://github.com/sharksforarms/deku/issues/76 :) ( :+1:  collaboration :100: )

TODO:
- attributes.rs docs update for bits/bytes
- possibly optimize the receiver, `bits = 4` or `bits = "4"` should both be ok (literal vs tokens)

### CODE REVIEW

File: deku-derive/src/lib.rs [hunk 1]
```diff
@@ -80,7 +80,25 @@ impl DekuData {
             .transpose()
             .map_err(|e| e.to_compile_error())?;
 
-        let bits = receiver.bytes.map(|b| b * 8).or(receiver.bits);
+        let bits = receiver
```
Comment (sharksforarms) [inline deku-derive/src/lib.rs:83]: Not the biggest fan of this section + unhanded unwrap

**Comment by sharksforarms** [comment]
I'm not convinced this is a best solution and your initial PR may be better. This branch doesn't use the "Limit" type but instead re-uses the bits and bytes. This means that we cannot pass this context to the T inside Vec<T> for example the following would no longer be possible:

3 elements with 5 bits each
```rust
#[deku(count = 3, bits = 5)]
Vec<u8>
```

**Comment by sharksforarms** [comment]
Given this exercise, I'm more inclined to add more attributes defining Limits on container types such as Vec<T>

**Comment by samuelsleight** [comment]
> This means that we cannot pass this context to the T inside Vec

Yeah, as much as I prefer the simplicity of just having the `"bits`" attribute instead of my current more verbose attempt, this was the main reason I opted for what I did.

I briefly also considered some weird "nested" `"bits"` attribute taking a comma separated list of values, such that `bits = "10,5"` would read 10 bits total into the vec, with 5 bits for each element, but I think this is far too obtuse, and there's too many questions as to how it would interact with having other attributes (such as `"count"`) or if you only want to pass one of the two values.

New attributes definitely feel like the best option, it just saves having to worry about weird interactions like that, and allows the existing attributes to work as intended

**Pull request closed without merging**